packages/opencensus-instrumentation-mongodb/test/test-mongodb.ts

```diff
@@ -0,0 +1,280 @@
+          assertSpan(
+              rootSpanVerifier, `${dbName}.${collectionName}.query`,
+              'db.mongodb.query');
+          done();
+        });
+      });
+    });
+
+    it('should create a child span for update', (done) => {
+      tracer.startRootSpan({name: 'updateRootSpan'}, (rootSpan) => {
+        collection.updateOne({a: 2}, {$set: {b: 1}}, (err, result) => {
+          assert.strictEqual(rootSpanVerifier.endedRootSpans.length, 0);
+          rootSpan.end();
+          assert.equal(err, null);
+          assertSpan(
+              rootSpanVerifier, `${dbName}.${collectionName}.query`,
+              'db.mongodb.query');
+          done();
+        });
+      });
+    });
+
+    it('should create a child span for remove', (done) => {
+      tracer.startRootSpan({name: 'removeRootSpan'}, (rootSpan) => {
+        collection.deleteOne({a: 3}, (err, result) => {
+          assert.strictEqual(rootSpanVerifier.endedRootSpans.length, 0);
+          rootSpan.end();
+          assert.equal(err, null);
+          assertSpan(
+              rootSpanVerifier, `${dbName}.${collectionName}.query`,
+              'db.mongodb.query');
+          done();
+        });
+      });
+    });
+  });
+
+  /** Should intercept cursor */
+  describe('Instrumenting cursor operations', () => {
+    it('should create a child span for find', (done) => {
+      tracer.startRootSpan({name: 'findRootSpan'}, (rootSpan) => {
+        collection.find({}).toArray((err, result) => {
+          assert.strictEqual(rootSpanVerifier.endedRootSpans.length, 0);
+          rootSpan.end();
+          assert.equal(err, null);
+          assertSpan(
+              rootSpanVerifier, `${dbName}.${collectionName}.cursor`,
+              'db.mongodb.query');
+          done();
+        });
+      });
+    });
+  });
+
+  /** Should intercept command */
+  describe('Instrumenting command operations', () => {
+    it('should create a child span for create index', (done) => {
+      tracer.startRootSpan({name: 'indexRootSpan'}, (rootSpan) => {
+        collection.createIndex({a: 1}, null, (err, result) => {
+          assert.strictEqual(rootSpanVerifier.endedRootSpans.length, 0);
+          rootSpan.end();
+          assert.equal(err, null);
+          assertSpan(
+              rootSpanVerifier, `${dbName}.$cmd.createIndexes`,
+              'db.mongodb.query');
+          done();
+        });
+      });
+    });
+
+    it('should create a child span for count', (done) => {
+      tracer.startRootSpan({name: 'countRootSpan'}, (rootSpan) => {
+        collection.count({a: 1}, (err, result) => {
+          assert.strictEqual(rootSpanVerifier.endedRootSpans.length, 0);
+          rootSpan.end();
+          assert.equal(err, null);
+          assertSpan(
+              rootSpanVerifier, `${dbName}.$cmd.count`, 'db.mongodb.query');
+          done();
+        });
+      });
+    });
+  });
+
+  /** Should intercept command */
+  describe('Removing Instrumentation', () => {
```

Comment:
Is it possible to make each test isolated? In other words, this test should still work (and be meaningful) if I replaced `describe` with `describe.only`.

Reply:
In the `beforeEach()` we insert sufficient data to run the tests independently of each other. Right now, it is possible to run with `describe.only()` or `describe.skip()` at any inner `describe` block.